src/load_tilejson.js

```diff
@@ -7,16 +7,40 @@ var request = require('./request'),
 module.exports = {
     _loadTileJSON: function(_) {
         if (typeof _ === 'string') {
-            _ = format_url.tileJSON(_, this.options && this.options.accessToken);
-            request(_, L.bind(function(err, json) {
-                if (err) {
-                    util.log('could not load TileJSON at ' + _);
-                    this.fire('error', {error: err});
-                } else if (json) {
-                    this._setTileJSON(json);
-                    this.fire('ready');
-                }
-            }, this));
+            if (_.indexOf('mapbox://') > -1) {
+                var styleURL = format_url.style(_, this.options && this.options.accessToken);
+                request(styleURL, L.bind(function(err, style) {
+                    if (err) {
+                        util.log('could not load Mapbox style at ' + styleURL);
+                        this.fire('error', {error: err});
+                    }
+                    var sources = [];
+                    for (var id in style.sources) {
+                        var source = style.sources[id].url.split('mapbox://')[1];
+                        sources.push(source);
+                    }
+                    request(format_url.tileJSON(sources.join(), this.options.accessToken), L.bind(function(err, json) {
```

Comment:
@tmcw  This is not making a request for every source, just one source. the tilejson endpoint supports concatenated sources, ex:

```
https://api.mapbox.com/v4/mapbox.mapbox-streets-v6,mapbox.satellite.json?access_token=pk.eyJ1IjoiYm9iYnlzdWQiLCJhIjoiTi16MElIUSJ9.Clrqck--7WmHeqqvtFdYig&secure
```